bundles/org.eclipse.swt/Eclipse SWT/common/org/eclipse/swt/graphics/SVGRasterizer.java

```diff
@@ -0,0 +1,47 @@
+/*******************************************************************************
+ * Copyright (c) 2024 Vector Informatik GmbH and others.
+ *
+ * This program and the accompanying materials are made available under the terms of the Eclipse
+ * Public License 2.0 which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors: Vector Informatik GmbH - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.swt.graphics;
+
+import java.io.*;
+
+/**
+ * Defines the interface for an SVG rasterizer, responsible for converting SVG
+ * data into rasterized images.
+ *
+ * @since 3.129
+ */
+public interface SVGRasterizer {
+
+	/**
+	 * Rasterizes an SVG image from the provided byte array, using the specified
+	 * zoom factor.
+	 *
+	 * @param stream        the SVG image as an {@link InputStream}.
+	 * @param scalingFactor the scaling ratio e.g. 2.0 for doubled size.
+	 * @return the {@link ImageData} for the rasterized image, or
+	 *         {@code null} if the input is not a valid SVG file or cannot be
+	 *         processed.
+	 * @throws IOException if an error occurs while reading the SVG data.
+	 */
+	public ImageData rasterizeSVG(InputStream stream, float scalingFactor) throws IOException;
```

Comment:
This method could return an optional, which would save us the separate `isSVGFile()` method.
```suggestion
	public Optional<ImageData> rasterizeSVG(InputStream stream, float scalingFactor) throws IOException;
```

Then, from a functionality point of view, this interface could be replaced by a 
`BiFunction<InputStream, Float, Optional<ImageData>>`

Reply:
I see your point. Right now the method is public and inside of `JSVGRasterizer` in the new plugin. Your suggestion is to make the method private and move it into `URLImageDescriptor` so it can be used in the two places there. Additionally it should be duplicated and placed into ImageLoader so it can be used in the `load` method. 

I don't have a stong opinion about this. @HannesWell what do you think?